mobile/src/screens/journal/journal.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import PlusIcon from '#assets/img/icons/plus.svg';
+import {
+  Card,
+  Header,
+  InputSearch,
+  LinearGradient,
+  Link,
+  ScrollView,
+  View,
+} from '#libs/components/components';
+import { AppColor } from '#libs/enums/enums';
+import {
+  useAppDispatch,
+  useAppRoute,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useNavigation,
+  useSearch,
+} from '#libs/hooks/hooks';
+import { actions as journalActions } from '#slices/journal/journal';
+
+import { styles } from './styles';
+
+const Journal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation();
+  const route = useAppRoute();
+
+  const { allJournalEntries } = useAppSelector(({ journal }) => {
+    return {
+      allJournalEntries: journal.allJournalEntries,
+    };
+  });
+  const badgeCount = allJournalEntries.length;
+
+  const { filteredData: filteredJournals, setSearchQuery } = useSearch(
+    allJournalEntries,
+    'title',
+  );
+
+  const handleSelectJournal = useCallback(() => {
+    // TODO: Implement actual functionality for the onPress event
+  }, []);
+
+  useEffect(() => {
+    navigation.setOptions({
+      header: () => {
+        return <Header title={route.name} badgeCount={badgeCount} />;
```

Comment:
```suggestion
        return <Header badgeCount={badgeCount} />;
```
Title is route name by default. You can also remove everything related to `useAppRoute` from this file